packages/@aws-cdk/aws-codedeploy/lib/ecs/deployment-group.ts

```diff
@@ -55,8 +212,135 @@ export class EcsDeploymentGroup {
     return new ImportedEcsDeploymentGroup(scope, id, attrs);
   }
 
-  private constructor() {
-    // nothing to do until CFN supports ECS deployment groups
+  public readonly application: IEcsApplication;
+  public readonly deploymentGroupName: string;
+  public readonly deploymentGroupArn: string;
+  public readonly deploymentConfig: IEcsDeploymentConfig;
+  /**
+   * The service Role of this Deployment Group.
+   */
+  public readonly role: iam.IRole;
+
+  private readonly alarms: cloudwatch.IAlarm[];
+
+  constructor(scope: Construct, id: string, props: EcsDeploymentGroupProps) {
+    super(scope, id, {
+      physicalName: props.deploymentGroupName,
+    });
+
+    this.application = props.application || new EcsApplication(this, 'Application');
+    this.alarms = props.alarms || [];
+
+    this.role = props.role || new iam.Role(this, 'ServiceRole', {
+      assumedBy: new iam.ServicePrincipal('codedeploy.amazonaws.com'),
+    });
+
+    this.role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AWSCodeDeployRoleForECS'));
+    this.deploymentConfig = props.deploymentConfig || EcsDeploymentConfig.ALL_AT_ONCE;
+
+    for (const service of props.services) {
+      if (cdk.Resource.isOwnedResource(service)) {
+        // Ensure the service's deployment controller is CodeDeploy
+        const cfnService = service.node.defaultChild as ecs.CfnService;
+        cfnService.deploymentController = { type: ecs.DeploymentControllerType.CODE_DEPLOY };
+
+        // Strip the revision ID from the service's task definition property to
+        // prevent new task def revisions in the stack from triggering updates
+        // to the stack's ECS service resource
+        const taskDef = (service as BaseService).taskDefinition;
+        cfnService.taskDefinition = taskDef.family;
+        service.node.addDependency(taskDef);
+      }
+    }
```

Comment:
I would feel more comfortable if this kind of behavior was configured on the ECS service (`managedByCodeDeploy: true` or what have you), and this code only validates that the service is configured correctly.

This feels like it can cause unexpected behavior.